client/src/components/img-button/img-button.js

```diff
@@ -0,0 +1,62 @@
+import Component from '@/components/component';
+
+import ImgBox from '@/components/img-box/img-box';
+import DefaultImg from '@/assets/images/button-default.svg';
+
+import styles from '@/styles/components/img-button/img-button.module.scss';
+
+const MAX_IMAGE_UPLOAD = 10;
+
+export default class ImgButton extends Component {
+  constructor(props) {
+    super(props);
+
+    this.$dom = this.createDom('div', {
+      className: styles['img-button'],
+    });
+
+    this.ImgBox = new ImgBox({
+      type: 'medium',
+      imageURL: '',
+      onClick: () => {},
+    });
+
+    this.render();
+    this.addEvent();
+  }
+
+  render = () => {
+    if (this._props.buttonType === 'add') {
+      this.$dom.innerHTML = `
+        <div class="${styles['ImgBox']}"></div>
+        <div class="${styles['content-wrapper']}">
+          <img src="${DefaultImg}">
+          <div class="${styles['image-count']}">${this._props.imageCount}/${MAX_IMAGE_UPLOAD}</div>
+        </div>
+      `;
+    } else {
+      this.$dom.innerHTML = `
+        <div class="${styles['ImgBox']}"></div>
+        <button class="${styles['remove-button']}">
+          <i class="wmi-close"></i>
+        </button>
+      `;
+    }
+
+    this.replaceElement(this.$dom.querySelector(`.${styles['ImgBox']}`), this.ImgBox.$dom);
+  };
+
+  addEvent = () => {
+    this.$dom.addEventListener('click', (e) => {
+      this._props.onClick();
+    });
+
+    if (this._props.buttonType === 'add') return;
+
+    const $removeButton = this.$dom.querySelector(`.${styles['remove-button']}`);
+    $removeButton.addEventListener('click', (e) => {
+      e.stopPropagation();
+      this._props.onRemove();
+    });
```

Comment:
```suggestion
    if (this._props.buttonType === 'add') {
      this.$dom.addEventListener('click', (e) => {
        this._props.onClickAdd();
      });
    } else if (this._props.buttonType === 'image') {
      const $removeButton = this.$dom.querySelector(`.${styles['remove-button']}`);
      $removeButton.addEventListener('click', (e) => {
        this._props.onClickRemove();
      });
    }
```

type이 add일 경우에는 onClickAdd만 필요하고 image일 경우는 onClickRemove만 필요한 상황이니 두개를 나눠서 처리하면 좋겠습니다.

중간에 if를 둬서 return 하셨는데 위에서 설명한 것과 같이 type을 확실하게 검사하는게 좋다고 생각합니다.
또한 type image의 경우는 그냥 이미지 클릭이 필요가 없으니 onClickRemove만 처리하면 좋겠습니다.

Reply:
코드 읽기가 훨씬 낫네요. 감사합니다.